pinecone/data/features/bulk_import.py

```diff
@@ -0,0 +1,193 @@
+                api_client_klass=ApiClient,
+                api_klass=BulkOperationsApi,
+                config=config,
+                openapi_config=openapi_config,
+                pool_threads=kwargs.get("pool_threads", 1),
+                api_version=API_VERSION,
+            )
+
+    def start_import(
+        self,
+        uri: str,
+        integration_id: Optional[str] = None,
+        error_mode: Optional[Literal["CONTINUE", "ABORT"]] = "CONTINUE",
+    ) -> StartImportResponse:
+        """Import data from a storage provider into an index. The uri must start with the scheme of a supported
+        storage provider. For buckets that are not publicly readable, you will also need to separately configure
+        a storage integration and pass the integration name.
+
+        Examples:
+            >>> from pinecone import Pinecone
+            >>> index = Pinecone().Index('my-index')
+            >>> index.start_import(uri="s3://bucket-name/path/to/data.parquet")
+            { id: "1" }
+
+        Args:
+            uri (str): The URI of the data to import. The URI must start with the scheme of a supported storage provider.
+            integration (Optional[str], optional): If your bucket requires authentication to access, you need to pass the name of your storage integration using this property. Defaults to None.
+            error_mode: Defaults to "CONTINUE". If set to "CONTINUE", the import operation will continue even if some
+                records fail to import. Pass "ABORT" to stop the import operation if any records fail to import.
+
+        Returns:
+            StartImportResponse: Contains the id of the import operation.
+        """
+        if isinstance(error_mode, ImportErrorMode):
+            error_mode = error_mode.value
+        elif isinstance(error_mode, str):
+            try:
+                error_mode = ImportErrorMode(error_mode.lower()).value
+            except ValueError:
+                raise ValueError(f"Invalid error_mode value: {error_mode}")
+
+        args_dict = parse_non_empty_args(
+            [
+                ("uri", uri),
+                ("integration_id", integration_id),
+                ("error_mode", ImportErrorModeClass(on_error=error_mode)),
+            ]
+        )
+
+        return self.__import_operations_api.start_import(StartImportRequest(**args_dict))
+
+    def list_imports(self, **kwargs) -> Iterator[List[ImportModel]]:
+        """
+        Returns a generator that yields each import operation. It automatically handles pagination tokens on your behalf so you can
+        easily iterate over all results. The `list_imports` method accepts all of the same arguments as list_imports_paginated
+
+        ```python
+        for op in index.list_imports():
+            print(op)
+        ```
+
+        You can convert the generator into a list by wrapping the generator in a call to the built-in `list` function:
```

Comment:
noice

Reply:
Since we bumped the version of Node and some of our TypeScript config stuff, we may be able to use generators in the TypeScript codebase.  🤔 

I had written one for the `List` endpoint but ended up not shipping it.